client/client.go

```diff
@@ -955,6 +896,12 @@ func (c *Client) updateNodeStatus() error {
 	defer c.heartbeatLock.Unlock()
 	c.lastHeartbeat = time.Now()
 	c.heartbeatTTL = resp.HeartbeatTTL
+
+	if err := c.rpcProxy.UpdateFromNodeUpdateResponse(&resp); err != nil {
```

Comment:
Pass in the individual arguments you need and if you come up with a better name change it